packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -0,0 +1,227 @@
+﻿/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type {
+  Informer,
+  KubernetesObject,
+  ListPromise,
+  ObjectCache,
+  V1Deployment,
+  V1Pod,
+} from '@kubernetes/client-node';
+import { AppsV1Api, CoreV1Api, KubeConfig, makeInformer } from '@kubernetes/client-node';
+import type { KubeContext } from './kubernetes-context.js';
+import type { ApiSenderType } from './api.js';
+
+// ContextInternalState stores informers for a kube context
+interface ContextInternalState {
+  podInformer?: Informer<V1Pod> & ObjectCache<V1Pod>;
+  deploymentInformer?: Informer<V1Deployment> & ObjectCache<V1Deployment>;
+}
+
+// ContextState stores information for the user about a kube context: is the cluster reachable, the number
+// of instances of different resources
+export interface ContextState {
+  reachable: boolean;
+  podsCount: number;
+  deploymentsCount: number;
```

Comment:
it's just questions as it's an internal object we can changes later, when we want

should it be podsCount or `{pods: {count : number}}` if we add other stuff related to pods later

about reachable, I think we should add the last time it was reachable as well

Reply:
IMHO it should be changed to `pods: PodInfo[]` in a future PR, ditto for deployments. Easy enough to get the count from there and can replace current-context informer.

Good call on reachable. We could track timestamp on every interaction so that when disconnected we know the last time it was connected, and when connected we know the last time we had an update or when it connected. IMHO this could wait though.